prompt_testing/README.md

```diff
@@ -0,0 +1,241 @@
+# Prompt Testing Framework
+
+A comprehensive framework for testing and iterating on prompts for the Claude explain service.
+
+## Overview
+
+This framework allows you to:
+- Test prompts against a curated set of assembly/source code examples
+- Compare different prompt versions automatically
+- Score responses using both automatic metrics and human review
+- Track performance over time and identify regressions
+
+## Quick Start
+
+```bash
+# List available test cases and prompts
+uv run prompt-test list
+
+# Run current prompt against basic optimization cases
+uv run prompt-test run --prompt current --categories basic_optimizations
+
+# Compare two prompt versions
+uv run prompt-test run --prompt v1_baseline --compare current
+
+# Analyze all previous results
+uv run prompt-test analyze
+```
+
+## Directory Structure
+
+```
+prompt_testing/
+├── test_cases/           # Test case definitions (JSON files)
```

Comment:
I'd prefer we use YAML here as it's slightly less horrible for humans to write and keep up to date (and trivial to turn into JSON)